test/TorchSharpTest/TestTorchTensor.cs

```diff
@@ -2192,12 +2192,16 @@ public void InfinityTest()
                 Assert.True(tensor.isinf().data<bool>().ToArray().All(b => b));
                 Assert.False(tensor.isneginf().data<bool>().ToArray().All(b => b));
                 Assert.False(tensor.isfinite().data<bool>().ToArray().All(b => b));
+                Assert.False(tensor.isnan().data<bool>().ToArray().All(b => b));
+                Assert.False(torch.isnan(tensor).data<bool>().ToArray().All(b => b));
 
                 tensor.fill_(Single.NegativeInfinity);
                 Assert.True(tensor.isneginf().data<bool>().ToArray().All(b => b));
                 Assert.True(tensor.isinf().data<bool>().ToArray().All(b => b));
                 Assert.False(tensor.isposinf().data<bool>().ToArray().All(b => b));
                 Assert.False(tensor.isfinite().data<bool>().ToArray().All(b => b));
+                Assert.False(tensor.isnan().data<bool>().ToArray().All(b => b));
```

Comment:
Should we have a unit test where isnan is true?

Reply:
> Should we have a unit test where isnan is true?

I've implemented a better test of `isnan`.